### PR TITLE
Adds choice for role and account

### PR DESCRIPTION
**Pull request by ricciocri**
We have multiple roles and accounts for each users and so it's not so practical to have multiple configuration files for each user.
With this change each user can choose which account/role assume.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     author='Cameron Marlow',
     author_email='cameron@healthcoda.com',
     url='https://github.com/healthcoda/onelogin-aws-cli',
-    download_url='https://github.com/healthcoda/onelogin-aws-cli/archive/0.1.2.tar.gz',
+    download_url='https://github.com/healthcoda/onelogin-aws-cli/archive/0.1.3.tar.gz',
```
Comment (cameronmarlow) [inline setup.py]: repo has changed to https://github.com/physera/onelogin-aws-cli

File: README.md [hunk 2]
```diff
@@ -18,5 +18,21 @@ To configure the script, simply run the configuration:
...
+OTP Token: 579114
+
+Please choose the role you would like to assume:
+[ 0 ]:  arn:aws:iam::166878887401:role/onelogin-test-ec2
+[ 1 ]:  arn:aws:iam::166878887401:role/onelogin-test-s3
+[ 2 ]:  arn:aws:iam::772123451421:role/onelogin-test-s3
+Selection:
+2
+Credentials cached in '/Users/myuser/.aws/credentials'
+Use aws cli with --profile 772123451421:role/onelogin-test-s3/myuser@mycompany.com
```
Comment (cameronmarlow) [inline README.md]: This seems out of scope for this package and quickly lost in console. Why isn't this your default role?

File: onelogin_aws_cli/__init__.py [hunk 3]
```diff
@@ -110,12 +115,64 @@ def get_saml_assertion(self):
...
+
+        for saml2attribute in root.iter('{urn:oasis:names:tc:SAML:2.0:assertion}Attribute'):
+             if (saml2attribute.get('Name') == 'https://aws.amazon.com/SAML/Attributes/Role'):
+                 for saml2attributevalue in saml2attribute.iter('{urn:oasis:names:tc:SAML:2.0:assertion}AttributeValue'):
+                     awsroles.append(saml2attributevalue.text)
+
+        # Note the format of the attribute value should be role_arn,principal_arn
+        # but lots of blogs list it as principal_arn,role_arn so let's reverse
+        # them if needed
+        for awsrole in awsroles:
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]: faster, simpler with list comprehensions:

```python
aws_roles = [role.split(",") for role in aws_roles]
aws_roles = [(principal, role) for role, principal in aws_roles]
```

Also, easier to store them as a tuple

File: onelogin_aws_cli/__init__.py [hunk 4]
```diff
@@ -110,12 +115,64 @@ def get_saml_assertion(self):
...
+
+    def get_role(self):
+        if not self.allroles:
+            self.get_arns()
+        # If I have more than one role, ask the user which one they want,
+        # otherwise just proceed
+        if len(self.allroles) > 1:
+            i = 0
+            print("Please choose the role you would like to assume:")
+            for awsrole in self.allroles:
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]:
```python
for i in range(0, len(all_roles):
    print("[{}] {}".format(i, aws_roles[i][0])
```

File: onelogin_aws_cli/__init__.py [hunk 5]
```diff
@@ -110,12 +115,64 @@ def get_saml_assertion(self):
...
+                print('[', i, ']: ', awsrole.split(',')[0])
+                i += 1
+
+            print ("Selection: ")
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]:
```python
selected_role = int(input("Selection: "))
```

File: onelogin_aws_cli/__init__.py [hunk 6]
```diff
@@ -110,12 +115,64 @@ def get_saml_assertion(self):
...
+            selectedroleindex = input()
+
+            # Basic sanity check of input
+            if int(selectedroleindex) > (len(self.allroles) - 1):
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]: Why not try/except:
```python
try:
    self.role_arn, self.principal_arn = self.all_roles(selected_role)
except IndexError:
    print("You selected an invalid role index, please try again")
    sys.exit(0)
```

**Comment by ricciocri** [comment]
Thanks for the review.
I'm really new in python and your output it's really useful to me.

Regarding the print of the profile as last line, in my company we have more than 10 aws accounts and access to multiple roles, so in the same day I could use a profile to have access to S£ on Dev account and after that give some commands on EC2 on production, the print of the line just help me a bit in remembering the name of the profile. 

I think could be useful if one has many accounts or roles.

I'll do the changes and push them to this PR.

Best Regards

**Comment by ricciocri** [comment]
Hello @cameronmarlow, I should have fix all the things you have commented, can you please review this new version ?

Best Regards

File: onelogin_aws_cli/__init__.py [hunk 7]
```diff
@@ -84,38 +88,78 @@ def get_saml_assertion(self):
...
+                    aws_roles.append(saml2attributevalue.text)
+
+        # Note the format of the attribute value should be role_arn,principal_arn
+        # but lots of blogs list it as principal_arn,role_arn so let's reverse
+        # them if needed
+        aws_roles = [role.split(",") for role in aws_roles]
+        aws_roles = [(role, principal ) for role, principal in aws_roles]
+        self.all_roles = aws_roles
+
+    def get_role(self):
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]: You're not currently handling the case where there is one role (self.principal_arn and self.role_arn aren't getting set).

Here's a working version:
```python
    def get_role(self):
        if not self.all_roles:
            self.get_arns()

        if not self.all_roles:
            raise Exception("No roles found")

        selected_role = None

        # If I have more than one role, ask the user which one they want,
        # otherwise just proceed
        if len(self.all_roles) > 1:
            for role, principal in self.all_roles:
                print("[{}] {}".format(role, principal))
            while not selected_role:
                choice = int(input("Role : "))
                if choice in range(len(self.all_roles)):
                    selected_role = choice
                else:
                    print("Invalid role index, please try again")
        else:
            selected_role = 0

        self.role_arn, self.principal_arn = self.all_roles[selected_role]
```

**Comment by ricciocri** [comment]
Hello @cameronmarlow thanks again for your suggestions, I should have addressed all the problems.
I've add a check if the directory ~/.aws exists and otherwise I create it.

Best Regards

File: onelogin_aws_cli/__init__.py [hunk 8]
```diff
@@ -121,38 +121,46 @@ def get_arns(self):
...
             self.get_arns()
+
+        if not self.all_roles:
+            raise Exception("No roles found")
+
+        selected_role = None
+
         # If I have more than one role, ask the user which one they want,
         # otherwise just proceed
+        ind = 0
```
Comment (cameronmarlow) [inline onelogin_aws_cli/__init__.py]: This should probably be inside the if scope below

**Comment by ricciocri** [comment]
FIxed :)
